paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PrepareResourcesTask.kt

```diff
@@ -105,7 +105,7 @@ abstract class PrepareResourcesTask : DefaultTask() {
         it.newLine()
         it.write(resourcePackageNames)
         it.newLine()
-        it.write(projectResourceDirs.joinFiles(projectDirectory))
+        it.write(projectResourceDirs.joinFiles(projectDirectory, reversed = true))
```

Comment:
Need to reverse the project res directory order compared to previous. Previously the list is in [ascending](https://cs.android.com/android-studio/platform/tools/base/+/mirror-goog-studio-main:build-system/gradle-core/src/main/java/com/android/build/gradle/api/BaseVariant.java;drc=5ac687029454dec1e7bd50697dabbc24d5a9943c;l=130) order, but in new api, it is in [descending](https://cs.android.com/android-studio/platform/tools/base/+/mirror-goog-studio-main:build-system/gradle-api/src/main/java/com/android/build/api/variant/SourceDirectories.kt;drc=5ac687029454dec1e7bd50697dabbc24d5a9943c;l=110) order. And we cannot reverse it in `PaparazziPlugin`, it is too early to do it as following error. Any suggestion for better place is appreciated.

```
> Could not create task ':preparePaparazziDebugResources'.
      > Querying the mapped value of list(interface java.util.Collection, org.gradle.api.internal.provider.AbstractCollectionProperty$CollectingSupplier@7576c856) before task ':generateDebugResValues' has completed is not supported
```

Reply:
an alternative might be:
```kotlin
val localResourceDirs = project
    .files(variant.sources.res!!.all).elements.map { it.reversed() }
```
or a probably simpler variant, keeping the FileCollection variable type (_I think this is the best one_):
```kotlin
val localResourceDirs = project
    .files(variant.sources.res!!.all.map { it.reversed() })
```
but the above only changes the first layer of lists (which might be enough!), not sure how reversed it needs to be, if a fully reversed list is needed:
```kotlin
val localResourceDirs = project
    .files(variant.sources.res!!.all.map { it.flatten().reversed() })
```

Reply:
never mind, I found a way to reverse the list in PaparazziPlugin, use `map` from `Provider`